modules/appeals_api/config/schemas/v2/200995.json

```diff
@@ -0,0 +1,164 @@
+                    }
+                  },
+                  "additionalProperties": false,
+                  "required": [ ]
+                },
+                "evidenceSubmission": {
+                  "type": "object",
+                  "properties": {
+                    "evidenceType":{
+                      "type": "array",
+                      "items": { "enum": ["uploaded", "retrieved"] },
+                      "minItems": 1,
+                      "uniqueItems": true
+                    },
+                    "retrieveFrom": {
+                      "type": "array",
+                      "items": {
+                        "type": "array",
+                        "items": {
+                          "type": "object",
+                          "properties": {
+                            "type": { "type": "string", "enum": ["retrievedEvidence"] },
+                            "attributes": {
+                              "type": "object",
+                              "properties": {
+                                "locationAndName": { "$ref": "#/definitions/nonBlankString" },
+                                "evidenceDates": {
+                                  "type": "array",
+                                  "items": { "$ref": "#/definitions/date" },
+                                  "minItems": 1
+                                }
+                              },
+                              "additionalProperties": false,
+                              "required": [ "locationAndName", "evidenceDates" ]
+                            }
+                          },
+                          "additionalProperties": false,
+                          "required": [ "type", "attributes" ]
+                        },
+                        "minItems": 1,
+                        "uniqueItems": true
+                      }
+                    }
+                  },
+                  "required": ["evidenceType"],
+                  "anyOf": [
+                    { "not": { "$ref": "#/definitions/retrieve-evidence" } },
+                    { "required": ["retrieveFrom"] }
+                  ],
+                  "definitions": {
+                    "retrieve-evidence": {
+                      "properties": {
+                        "evidenceType": { "contains": { "enum": ["retrieved"] } }
+                      }
```

Comment:
this is a sub definition called implication, that basically ensures on a sub-schema level that if `retrieved` is provided, there's also locations provided.

Reply:
[Implication Docs](https://json-schema.org/understanding-json-schema/reference/conditionals.html#implication)

Reply:
Is the `$ref` path for the "not" correct, though? It seems like it should be `#/definitions/scCreate/definitions/retrieve-evidence`, since it's a sub-definition? Or are all definitions bubbled up into the `#/definitions` root namespace?

Reply:
I followed the docs when writing this, so i'm just assuming that it's the latter.